sqlmesh/core/reference.py

```diff
@@ -0,0 +1,113 @@
+from __future__ import annotations
+
+import typing as t
+from collections import defaultdict, deque
+
+from pydantic import validator
+from sqlglot import exp
+
+from sqlmesh.utils.errors import ConfigError, SQLMeshError
+from sqlmesh.utils.pydantic import PydanticModel
+
+if t.TYPE_CHECKING:
+    from sqlmesh.core.model import Model
+
+
+class Reference(PydanticModel, frozen=True):
+    expression: exp.Expression
+    dialect: str
```

Comment:
How is this dialect used?

Reply:
used for the parsing of the expression in the serialized form i think

Reply:
Really? Where?

Reply:
in the validator, parse_expression